### DISCUSSION TITLE
Non-Null Capture when comparing Is Null

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#5788] [Language Ideas]
Today, you can write the code:
```
if(a?.b?.c?.d is {} V1) {
  // ... V1 is definitely assigned to not null
}
``

It would be nice if you could also do the following:
```
if(a?.b?.c?.d is null V1){
  // ... V1 is definitely null
} else {
  // ... V1 is not null  
}
```

**Comment by jnm2** [comment]
Today you can also write `is not null and var v1` for reference types. This is decidedly unpleasant with nullable value types. @CyrusNajmabadi once mentioned interest in shortening this, but I'm not sure if he's still interested.

I still write `is { } v1` and remain unconvinced that this syntax is more confusing than anything else someone hasn't gotten used to.

**Reply by CyrusNajmabadi** [reply]
i'm not really that interested anymore.  i think the alternatives are fine :)

**Comment by RikkiGibson** [comment]
```cs
if (x is not {} y)
{
    // we don't have a y
}
else
{
    // we do have a y
}
```
